pkg/activator/net/throttler.go

```diff
@@ -411,9 +411,8 @@ func (rt *revisionThrottler) handleUpdate(update revisionDestsUpdate) {
 		zap.String("ClusterIP", update.ClusterIPDest), zap.Object("dests", logging.StringSet(update.Dests)))
 
 	// ClusterIP is not yet ready, so we want to send requests directly to the pods.
-	// NB: this will not be called in parallel, thus we can build a new podTrackers
```

Comment:
I think `ClusterIPDest` is explicitly set to empty string:

https://github.com/knative/serving/blob/71088a8c74e811b769cb6bfd4c51e2021c8dc858/pkg/activator/net/revision_backends.go#L351-L354

to guarantee here to run without Lock. "If" the logic has problem, it is better to fix the logic instead of adding the Lock?